### PR TITLE
add app vendor management

### PR DESCRIPTION
**Pull request by huojiao2006**

### CODE REVIEW

File: api/appvendor.proto [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+service AppVendorManager {
+
+	rpc SubmitVendorVerifyInfo (SubmitVendorVerifyInfoRequest) returns (SubmitVendorVerifyInfoResponse) {
+		option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+			summary: "SubmitVendorVerifyInfo"
+		};
+		option (google.api.http) = {
+            post: "/v1/vendor_verify_infos/user_id=*",
```
Comment (zheng1) [inline api/appvendor.proto]: `post: "/v1/vendor_verify_infos/{user_id=*}",`

File: pkg/constants/common.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ const (
 	MarketManagerHost = prefix + "market-manager"
 
 	AttachmentManagerHost = prefix + "attachment-manager"
+	VendorManagerHost     = prefix + "appvendor-manager"
```
Comment (zheng1) [inline pkg/constants/common.go]: `vendor-manager` will be better

File: pkg/gerr/message.go [hunk 3]
```diff
@@ -242,6 +242,11 @@ var (
 		en:   "illegal URL format [%s]",
 		zhCN: "非法的URL格式[%s]",
 	}
+	ErrorIllegalLabelFormat = ErrorMessage{
```
Comment (zheng1) [inline pkg/gerr/message.go]: useless define

File: pkg/gerr/message.go [hunk 4]
```diff
@@ -367,4 +372,19 @@ var (
...
+	}
+	ErrorIllegalPhoneFormat = ErrorMessage{
+		Name: "illegal_phone_format",
+		en:   "illegal Phone format [%s]",
+		zhCN: "非法的Phone格式[%s]",
+	}
+	ErrorIllegalBankAccountNumberFormat = ErrorMessage{
+		Name: "illegal_bankAccountNumber_format",
+		en:   "illegal BankAccountNumber format [%s]",
+		zhCN: "非法的BankAccountNumber格式[%s]",
```
Comment (zheng1) [inline pkg/gerr/message.go]: need translate

File: pkg/models/vendor_verify_info.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/pb"
+	"openpitrix.io/openpitrix/pkg/util/pbutil"
+)
+
+const (
```
Comment (zheng1) [inline pkg/models/vendor_verify_info.go]: define in `pkg/constants/column.go`

File: pkg/service/appvendor/handler.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+// that can be found in the LICENSE file.
+
+package appvendor
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/manager"
+
```
Comment (zheng1) [inline pkg/service/appvendor/handler.go]: empty line

File: pkg/service/appvendor/validation.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	result := reg.MatchString(bankAccountNumberStr)
+	if result {
+		return true, nil
+	} else {
+		return false, gerr.New(ctx, gerr.InvalidArgument, gerr.ErrorValidateFailed, bankAccountNumberStr)
+	}
+}
+
+//Status
+func VerifyStatus(ctx context.Context, strStatus ...string) (bool, error) {
```
Comment (zheng1) [inline pkg/service/appvendor/validation.go]: use `Checker StringChosen`

File: pkg/service/appvendor/handler.go [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+		vendor = &models.AppVendor{}
+		vendor.UserId = userID
+		vendor.Status = models.StatusNew
+		vendor.StatusTime = time.Now()
+		userID, err = CreateVendorVerifyInfo(ctx, *vendor)
+		if err != nil {
+			logger.Error(ctx, "vendorVerifyInfo does not exit,create new vendorVerifyInfo failed [%s], %+v", userID, err)
+			return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorCreateResourceFailed)
+		}
+		logger.Debug(ctx, "vendorVerifyInfo does not exit,create new vendorVerifyInfo verify info,[%+v]", vendor)
```
Comment (zheng1) [inline pkg/service/appvendor/handler.go]: Space is required after commas

File: pkg/service/appvendor/resource_control.go [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+// that can be found in the LICENSE file.
+
+package appvendor
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/logger"
+
```
Comment (zheng1) [inline pkg/service/appvendor/resource_control.go]: empty line

File: pkg/models/vendor_verify_info.go [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+	TableVendor = "vendor_verify_info"
```
Comment (zheng1) [inline pkg/models/vendor_verify_info.go]: `TableVendorVerifyInfo`

File: test/appvendor/appvendor_test.go [hunk 11]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The OpenPitrix Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+// +build k8s
```
Comment (zheng1) [inline test/appvendor/appvendor_test.go]: should remove this line

File: pkg/models/vendor_verify_info.go [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+	ColumnSubmitTime        = "submit_time"
+	ColumnStatusTime        = "status_time"
+
+	StatusSubmitted = "submitted"
+	StatusNew       = "new"
+	StatusPassed    = "passed"
+	StatusRejected  = "rejected"
+)
+*/
+type AppVendor struct {
```
Comment (zheng1) [inline pkg/models/vendor_verify_info.go]: VendorVerifyInfo

File: pkg/service/appvendor/handler.go [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+	vendor, err := GetVendorVerifyInfo(ctx, userID)
+
+	if vendor == nil && err != nil {
+		vendor = &models.AppVendor{}
+		vendor.UserId = userID
+		vendor.Status = constants.StatusNew
+		vendor.StatusTime = time.Now()
+		userID, err = CreateVendorVerifyInfo(ctx, *vendor)
+		if err != nil {
+			logger.Error(ctx, "vendorVerifyInfo does not exit,create new vendorVerifyInfo failed [%s], %+v", userID, err)
```
Comment (zheng1) [inline pkg/service/appvendor/handler.go]: space after commas

File: pkg/service/appvendor/handler.go [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+			return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorCreateResourceFailed)
+		}
+		logger.Debug(ctx, "vendorVerifyInfo does not exit,create new vendorVerifyInfo verify info, [%+v]", vendor)
```
Comment (zheng1) [inline pkg/service/appvendor/handler.go]: space after commas

File: pkg/service/appvendor/handler.go [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+		}
+	} else {
+		vendor := &models.AppVendor{}
+		vendor = vendor.ParseReq2Vendor(req)
+		userID, err = CreateVendorVerifyInfo(ctx, *vendor)
+		if err != nil {
+			logger.Error(ctx, "Failed to submit vendorVerifyInfo [%+v], %+v", vendor, err)
+			return nil, gerr.NewWithDetail(ctx, gerr.Internal, err, gerr.ErrorCreateResourcesFailed)
+		}
+		logger.Debug(ctx, "vendorVerifyInfo does not exit,create new vendorVerifyInfo verify info,[%+v]", vendor)
```
Comment (zheng1) [inline pkg/service/appvendor/handler.go]: space after commas

File: pkg/service/appvendor/handler.go (hunk 6 above)
Comment (zheng1) [reply pkg/service/appvendor/handler.go]: line 12 should be removed

File: pkg/gerr/message.go (hunk 3 above)
Comment (huojiao2006) [reply pkg/gerr/message.go]: update

File: pkg/gerr/message.go (hunk 4 above)
Comment (huojiao2006) [reply pkg/gerr/message.go]: update

File: pkg/constants/common.go (hunk 2 above)
Comment (huojiao2006) [reply pkg/constants/common.go]: update

File: pkg/service/appvendor/handler.go (hunk 6 above)
Comment (huojiao2006) [reply pkg/service/appvendor/handler.go]: update

File: pkg/service/appvendor/validation.go (hunk 7 above)
Comment (huojiao2006) [reply pkg/service/appvendor/validation.go]: update

File: pkg/service/appvendor/handler.go (hunk 8 above)
Comment (huojiao2006) [reply pkg/service/appvendor/handler.go]: update

File: test/appvendor/appvendor_test.go (hunk 11 above)
Comment (huojiao2006) [reply test/appvendor/appvendor_test.go]: update

File: pkg/models/vendor_verify_info.go (hunk 12 above)
Comment (huojiao2006) [reply pkg/models/vendor_verify_info.go]: update

File: pkg/service/appvendor/handler.go (hunk 14 above)
Comment (huojiao2006) [reply pkg/service/appvendor/handler.go]: update

File: pkg/service/appvendor/handler.go (hunk 15 above)
Comment (huojiao2006) [reply pkg/service/appvendor/handler.go]: update

**Pull request closed without merging**